src/components/QueryEditor.tsx

```diff
@@ -1,59 +1,31 @@
 import React, { PureComponent } from 'react';
 import defaults from 'lodash/defaults';
-import { BracesPlugin, QueryField, Select, Field, Input, Collapse } from '@grafana/ui';
+import { QueryField, Select, Field, Input, Collapse } from '@grafana/ui';
 import { QueryEditorProps, SelectableValue } from '@grafana/data';
-import { DataSource } from './datasource';
-import { defaultQuery, LightstepDataSourceOptions, LightstepQuery } from './types';
+import { DataSource } from '../datasource';
+import { LightstepDataSourceOptions, LightstepQuery } from '../types';
 
 type Props = QueryEditorProps<DataSource, LightstepQuery, LightstepDataSourceOptions>;
-interface QueryEditorState {
-  errorMessage: string;
-  projects: string[];
```

Comment:
Data ownership clarification: Duplicate copy of datasource project state removed, the list of projects is requested from the datasource directly during render.